### PR TITLE
zk v0.9.0 (new formula)

### PR DESCRIPTION
**Pull request by mhanberg**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Provides a formula for [zk](https://github.com/mickael-menu/zk), a command-line tool helping you to maintain a plain text Zettelkasten or personal wiki.

### CODE REVIEW

**Comment by chenrui333** [comment]
```
  # github.com/mattn/go-sqlite3
  sqlite3-binding.c:198680:10: fatal error: 'unicode/ucol.h' file not found
```

**Comment by mhanberg** [comment]
`otools` is telling me the version i have working is linking to what I assume to be the macos version.

```
 otool -L $(which zk)
/opt/homebrew/bin/zk:
        /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1853.0.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.0.0)
        /opt/homebrew/opt/icu4c/lib/libicuuc.69.dylib (compatibility version 69.0.0, current version 69.1.0)
        /opt/homebrew/opt/icu4c/lib/libicui18n.69.dylib (compatibility version 69.0.0, current version 69.1.0)
        /System/Library/Frameworks/Security.framework/Versions/A/Security (compatibility version 1.0.0, current version 60157.30.13)
```

**Comment by SMillerDev** [comment]
> I think I need to compile with the homebrew version of icu4c, but the audit is telling me that i need to use the version bundled in macOS.

Maintainers can decide to ignore an audit if the software does not work otherwise. Just make it depend on the homebrew version.

> otools is telling me the version i have working is linking to what I assume to be the macos version.

`/opt/homebrew/opt/icu4c/` suggests that this is the homebrew version, not the the macOS native version.

**Comment by mhanberg** [comment]
> Maintainers can decide to ignore an audit if the software does not work otherwise. Just make it depend on the homebrew version.

Oh, gotcha. I'm not sure how to make the CI continue tho while failing the audit.

> /opt/homebrew/opt/icu4c/ suggests that this is the homebrew version, not the the macOS native version.

That was a typo, I meant the homebrew version 😅.

---

I'll push it up depending on the homebrew version.

**Review by carlocab — DISMISSED** [review]

File: Formula/zk.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  head "https://github.com/mickael-menu/zk.git", branch: "main"
+
+  depends_on "go" => :build
+
+  depends_on "icu4c"
+  uses_from_macos "sqlite"
+
+  def install
+    system "go", "build", *std_go_args, "-tags", "fts5,icu", "-ldflags", "-X=main.Version=#{version} -X=main.Build=brew"
```
Comment (SMillerDev) [inline Formula/zk.rb:16]: One more comment
```suggestion
    system "go", "build", *std_go_args(ldflags: "-X=main.Version=#{version} -X=main.Build=#{tap.user}"), "-tags", "fts5,icu"
```

**Review by SMillerDev — APPROVED** [review]

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**